src/codeu/chat/client/core/View.java

```diff
@@ -45,6 +46,23 @@ public View(ConnectionSource source) {
     this.source = source;
   }
 
+  public ServerInfo getInfo() {
+    try (final Connection connection = this.source.connect()) {
+      Serializers.INTEGER.write(connection.out(), NetworkCode.SERVER_INFO_REQUEST);
+      if (Serializers.INTEGER.read(connection.in()) == NetworkCode.SERVER_INFO_RESPONSE) {
+      final Uuid version = Uuid.SERIALIZER.read(connection.in());
+      return new ServerInfo(version);
+      } else {
+        LOG.error("Response from server failed.");
+      }
+    } catch (Exception ex) {
+      System.out.println("ERROR: Exception during call on server. Check log for details.");
+      LOG.error(ex, "Exception during call on server.");
+    }
+
+  return null;
```

Comment:
It's unfortunate that the told you to return null here. This is usually considered bad practice. The preferred way to model an optional return value is to use [java.util.Option](https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html)